resilience4j-bulkhead/src/main/java/io/github/resilience4j/bulkhead/adaptive/internal/config/AbstractConfig.java

```diff
@@ -0,0 +1,212 @@
+/*
+ *
+ *  Copyright 2019: Mahmoud Romeh
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ *
+ */
+package io.github.resilience4j.bulkhead.adaptive.internal.config;
+
+import java.time.Duration;
+import java.util.Objects;
+
+import io.github.resilience4j.bulkhead.BulkheadConfig;
+import io.github.resilience4j.core.lang.NonNull;
+
+/**
+ * abstract common configuration
+ */
+public class AbstractConfig {
```

Comment:
I think we can move this into `AdaptiveBulkheadConfig`.
The way how we measure metrics is not changable by the users. We provide to options: count-based and time-based. Maybe we add a third in the future, but that's up to us.

Reply:
Can you please explain more what you want here :) ? as Abstract config is our way to impose what is the current common configuration and metrics so for different policies ,users will just extend that one like we did for AIMD policy config class and add their extra properties if needed

Reply:
Yes, but the common configuration can be part of `AdaptiveBulkheadConfig`, or? 
The policy gets a `AdaptiveBulkheadConfig<WhatEverLimiter>` injected, or?

Reply:
I'm not talking about the class itself, but the content of the class can be moved into the common AdaptiveBulkheadConfig.

Reply:
I would like to keep it like that as it is as merging the content of AsbtractConfi into AdaptiveBulkHeadConfig will make the class too big and not easy to maintain , like that for now it is well encapsulated